docs/swarm/admin_guide.md

```diff
@@ -0,0 +1,243 @@
+- Reboot the machine.
+- If neither restarting or rebooting work, you should add another manager node or promote a worker to be a manager node. You also need to cleanly remove the failed node entry from the Manager set with `docker node demote <id-node>` and `docker node rm <id-node>`.
+
+Alternatively you can also get an overview of the cluster health with `docker node ls`:
+
+```bash
+# From a Manager node
+docker node ls
+ID                           HOSTNAME  MEMBERSHIP  STATUS  AVAILABILITY  MANAGER STATUS
+1mhtdwhvsgr3c26xxbnzdc3yp    node05    Accepted    Ready   Active
+516pacagkqp2xc3fk9t1dhjor    node02    Accepted    Ready   Active        Reachable
+9ifojw8of78kkusuc4a6c23fx *  node01    Accepted    Ready   Active        Leader
+ax11wdpwrrb6db3mfjydscgk7    node04    Accepted    Ready   Active
+bb1nrq2cswhtbg4mrsqnlx1ck    node03    Accepted    Ready   Active        Reachable
+di9wxgz8dtuh9d2hn089ecqkf    node06    Accepted    Ready   Active
+```
+
+## Manager advertise address
+
+When initiating or joining a Swarm cluster, you have to specify the `--listen-addr`
+flag to advertise your address to other Manager nodes in the cluster.
+
+We recommend that you use a *fixed IP address* for the advertised address, otherwise
+the cluster could become unstable on machine reboot.
+
+Indeed if the whole cluster restarts and every member gets a new IP address on
+restart, there is no way for any of those nodes to contact an existing member
+and the cluster will stay stuck tryiing to contact past members. While having
+dynamic IP addresses for Worker nodes is acceptable, Managers are meant to
+be a stable piece in the infrastructure thus it is highly recommended to deploy
+those critical nodes with static IPs.
+
+## Recover from disaster
+
+Swarm is resilient to failures and the cluster can recover from any number
+of temporary node failures (machine reboots or crash with restart).
+
+In a swarm of `N` members, there must be a quorum of manager nodes greater than
+50% of the total number of managers (`(N/2)+1`) in order for the swarm to
+process requests and remain available. This means the swarm can tolerate up to
+`(N-1)/2` permanent failures beyond which requests involving cluster management
+cannot be processed. These types of failures include data corruption or hardware
+failures.
+
+Even if you follow the guidelines here, it is possible that you can lose a
+quorum of manager nodes. If you can't recover the quorurm by conventional
+means such as restarting faulty nodes, you can recover the cluster by running
+`docker swarm init --force-new-cluster` on a manager node.
+
+```bash
+# From the node to recover
+docker swarm init --force-new-cluster --listen-addr node01:2377
+```
+
+The `--force-new-cluster` flag puts the Docker Engine into swarm mode as a
+manager node of a single-node cluster. It discards cluster membership information
+that existed before the loss of the quorum.
```

Comment:
Mention that it retains data such as services, tasks, and the list of worker nodes.